js/anims-missiles.js

```diff
@@ -59,41 +64,69 @@ function createMissileClickHandler() {
     var cities = document.querySelectorAll('.city')
     // forEach loop changed with Array.prototype.forEach.call due to compatibility issues in IE10
     Array.prototype.forEach.call(cities, function(city) {
-        city.addEventListener('click', function() {
-            columnID = this.getAttribute('data-city')
-            isHit()
-        })
-    })//end forEach
+        city.addEventListener('click', checkForMissiles)
+    })
 }
+
+/**
+ * This was previously an anonymous function given its own function. It sets the column id and executes the isHit()
+ * method to scan for any missiles in the column
+ */
+function checkForMissiles() {
+    columnID = this.getAttribute('data-city')
+    $city = $('#city-' +  columnID)
+    shootLaser($city)
+    isHit()
+}
+
 /**
  * If a missile is hit stop the anim and replace the missile img with explosion img - increment the deflects score.  If it is a miss decrement the deflects score and render on the page
  */
 function isHit() {
     var missileNumber = '.missile-' + columnID
     var missiles = document.querySelectorAll(missileNumber)
     var hasMissiles = missiles.length
+    var success = true
```

Comment:
why have this variable? you can simplify your code by removing this and just calling makeNoise in the relevant if, then you dont need the `laserResultSound` function

Reply:
this function is called both during the demo phase and during the actual game phase. We needed to overcome an earlier design desicion and this was the best way we could come up with in the time we had.